backend/satellite_tools/reposync.py

```diff
@@ -1130,13 +1130,20 @@ def import_packages(self, plug, source_id, url, is_non_local_repo):
             log(0, '')
             log(0, "  Linking packages to the channel.")
             # Packages to append to channel
-            import_batch = [self.associate_package(pack) for (pack, to_download, to_link) in to_process if to_link]
-            backend = SQLBackend()
-            caller = "server.app.yumreposync"
-            importer = ChannelPackageSubscription(import_batch,
-                                                  backend, caller=caller, repogen=False)
-            importer.run()
-            backend.commit()
+            import_batches = list(self.chunks(
+                [self.associate_package(pack) for (pack, to_download, to_link) in to_process if to_link],
+                1000))
```

Comment:
I'm wondering if 1000 is the best size for those chunks, I suggest we do some performance test in a future version

Reply:
Just to have the numbers from my test system:

```
# start: 2 Minutes
# Download: ~ 2 hours
# Import: ~ 5:30 hours
# Linking: 2 minutes
```

I think we do not have performance problems in the "linking" area.